xonsh/built_ins.py

```diff
@@ -217,14 +217,29 @@ def ensure_list_of_strs(x):
     return rtn
 
 
-def list_of_strs_or_callables(x):
-    """Ensures that x is a list of strings or functions"""
+def ensure_str_or_callable(x):
+    """Ensures that x is single string or function."""
     if isinstance(x, str) or callable(x):
-        rtn = [x]
+        return x
+    if isinstance(x, bytes):
+        # ``os.fsdecode`` decodes using "surrogateescape" on linux and "strict" on windows.
+        # This is used to decode bytes for interfacing with the os, notably for command line arguments.
+        # See https://www.python.org/dev/peps/pep-0383/#specification
+        return os.fsdecode(x)
```

Comment:
This should probably respect  `$XONSH_ENCODING` and `$XONSH_ENCODING_ERRORS`, rather than the system values https://xon.sh/envvars.html#xonsh-encoding

Reply:
This encoding is specifically meant for passing bytes to and from the os, and in the end the OS receives the exact same bytes. 
So it's not really a regular encoding, it will just pass the bytes to the os.